### DISCUSSION TITLE
Turbulent Channel Flow

### DISCUSSION THREAD

**Discussion by Kamyarmvn** [ExtremeFLOW/neko#1349] [General]
Hi,
I am trying to simulate turbulent flow inside a channel at Re_tau=180. The available mesh has 18 elements in each direction. I tried to generate a finer mesh with genmeshbox but then the transition to turbulence does not occur and the flow somehow dampens. 
![u velocity](https://github.com/ExtremeFLOW/neko/assets/161873038/2e4cb67b-fa88-455e-9676-3796d07327c4)
I noticed that when I create a mesh with the same number of elements, this problem does not appear. If I only change the mesh distribution in y direction there will be no problem too. Only when I try to increase mesh elements this problem appears. 
I would like to know others' opinion in this regard. What could be the cause of this? I look forward to any suggestion. 
best,
Kamyar

**Comment by Shiyu-Sandy-Du** [comment]
Did you change anything in the user file?

**Comment by Kamyarmvn** [comment]
I changed the mesh distribution in the wall normal direction to a smooth tanh function. This works with the original mesh but so far did not worked with any other mesh.

**Reply by Shiyu-Sandy-Du** [reply]
Do you have a visualization of v velocity?

**Reply by Shiyu-Sandy-Du** [reply]
I mean the time evolution of it

**Reply by Kamyarmvn** [reply]
u velocity after 60 seconds:
![u velocity](https://github.com/ExtremeFLOW/neko/assets/161873038/74496e9f-5d71-4e65-a24a-2a93f21ddded)
v velocity after 60 seconds:
![v velocity](https://github.com/ExtremeFLOW/neko/assets/161873038/5f7c44a4-7aa4-46d0-9e49-696f8ab8c9eb)
This is from a mesh size 24x18x20 and N=8, Re =2800. Domain size:4pi x 2delta x 4/3pi delta. The solution continues to damp. I abort the simulation after a while.

**Reply by Kamyarmvn** [reply]
The problem is I could not reproduce the results of Moser, Kim, and Mansour with the original mesh (Reynolds stresses were underestimated).

**Reply by Shiyu-Sandy-Du** [reply]
It makes sense if you cannot reproduce since your case is not turbulent by far. However, can you show a snapshot of the 18^3 case when it transition towards turbulence?

**Reply by Shiyu-Sandy-Du** [reply]
And maybe another thought: could you please try a different disturbance wavenumber? Like kx=7 and kz=5?

**Reply by Kamyarmvn** [reply]
Here is a snapshot of u velocity at 160s for the 18^3 mesh.
![u velocity](https://github.com/ExtremeFLOW/neko/assets/161873038/7bb77bbc-cad9-4e1b-b96b-6c7eac8d0d56)
I think the problem lies in the discrepancy between the mesh size and the user file. The actual mesh has a size of 3pi/2 in the spanwise direction instead of the 4pi/3 that is written in the file. I created meshes with the same size as the original one and so far it seems to be working. I think the initial condition is not suitable for a domain with a size of 4pi/3 in spanwise direction. Maybe a different disturbance wave number would work. I will try it out next. 
The mean values of the simulation were fairly satisfactory, only for Reynolds stresses there was a difference. I hope by increasing the mesh size I can improve the results. I am trying with 22 x 24 x 20 mesh now.

**Reply by Shiyu-Sandy-Du** [reply]
By "transition towards turbulence", I mean before 50s or even 20s etc when you see your initial disturbances begin to break down :)

**Reply by Kamyarmvn** [reply]
Sorry, I thought you meant around the end of the transition.:)
![u velocity](https://github.com/ExtremeFLOW/neko/assets/161873038/fc29b12b-82c0-4062-9c8a-348e22de9cf7)
This is at 20s.

**Reply by Shiyu-Sandy-Du** [reply]
Oh thanks for the screenshot. Could you also plot the contour with option "Surface with edges"? Could it be the irregular pattern close to the upper wall to be mesh-related?

**Reply by Shiyu-Sandy-Du** [reply]
Or could you please have a slice in paraview showing the elementary interfaces together with the contour you show here?

**Reply by Kamyarmvn** [reply]
![edge](https://github.com/ExtremeFLOW/neko/assets/161873038/346296c9-67f5-48cf-8697-6a681872a7b7)
I am not sure if you meant this or not.
![1](https://github.com/ExtremeFLOW/neko/assets/161873038/a9b378c1-85e3-4de1-a72b-6aab751e3932)
This one is from the original mesh. I am currently running a simulation on a finer mesh. Once it is finished I can say if the problem was the mesh size or not. There is a possibility that the difference in size in spanwise direction is the cause of difference.

**Reply by Shiyu-Sandy-Du** [reply]
Yes I mean this one. Could you also show a plot of xy plane? (I think this discussion just help us to understand why the transition does not happen, but for a more practical purpose, as Timofey says, the mesh-to-mesh interpolation from an already-turbulent case is better. :)

**Reply by Kamyarmvn** [reply]
![1](https://github.com/ExtremeFLOW/neko/assets/161873038/0262760f-75c1-4bbf-bb6b-f639041969f8)
I must say this is from the original mesh that does transition to turbulence. I have a couple of new cases with different sizes and revised initial case and domain size that seem to be working so far. 
I will check the mesh-to-mesh interpolation feature. thank you for the insight.:)

**Comment by Kamyarmvn** [comment]
I think there is discrepancy here. t seems that the original mesh has a different size in z direction. although in turbulent_channel.f90 file the size in z direction is said to be 4/3 pi. The initial condition is also based on a size of 4/3pi in z direction. 
Does anyone have a suggestion?
Thank you.

**Reply by timofeymukha** [reply]
I don't think these ICs have in general been very thought through, but just worked for the concrete case in the example. I think the simplest thing to do is run the example for a bit, get a turbulent field, and the use mesh-to-mesh interpolation to start out from a turbulent field in your new simulation.

**Reply by Kamyarmvn** [reply]
Thank you for the advice. I wasn't aware of the mesh-to-mesh interpolation. I look it up.

**Comment by pschlatt1** [comment]
I looked at the code for the initial condition:
      alpha = kx * 2*PI/(pi*4)
      beta  = kz * 2*PI/(pi*4/3)
      
So of course, it would be important that the 4pi and 4/3pi correpsond to the actual domain size used; if not you will have a discontinuity in the domain which is never good. So make sure that the alpha and beta are the correct fundamental wavenumbers for your case.

Then we select one wavenumber.
      eps = 3e-2
      kx  = 16
      kz  = 7
Now the choice is usually not very sensitivte, but if you want to reproduce the "same" ininitial condition with a different mesh, then you may need to re-compute the wavenumbers, i.e. to make sure you have the same wavelength. So for instance:
lambda= 2pi/(beta_old*kz_old); kz_new = 2pi/lambda/beta_new  = 2pi/2pi*beta_old*kz_old/beta_new  -->  kz_new=kz_old*(beta_old/beta_new).

I don't like this initial condition very much, and I would add a bit of random noise on top of that as well. You can create some randomness by coming up wiht some carzy function of coordinates, e.g. random = sin(x**2-500*tan(y**3-2*z)+100*zz**3*x+exp(x)) or so, and just add it to the velcoity (with a scale of e.g. 1e-2). So the basic idea is to have a low frequency disturbance, and a high-requency noise on top.

**Reply by Kamyarmvn** [reply]
Thank you for the insight. Yes, the initial condition corresponds to a mesh with a size of 4pi and 4pi/3 in streamwise and spanwise direction. The interesting part was that at least I couldn't get a transition to turbulence with a mesh of the same size. The actual mesh used in the test case has a 3pi/2 size in spanwise direction and interestingly, the initial condition works for it. I created a new mesh with a size of 3pi/2 in z direction and altered the beta to kz*2*pi/(3pi/2) and it seems to be working so far. 
I will run a case based on the new initial condition and check the results. 

Thank you very much.

**Comment by pschlatt1** [comment]
ok, I will try to run the case in the examples directory, and see what the problem could be. I suggest that we stick to the sizes previously used. KMM87 had 4pi x 2pi, and MKM99 had 4pi x 4/3 pi for the low Re, and 2pi x pi for the higher Reynolds numbers. 

I think the length in x should be 4pi, and I will test for 4/3pi and 2pi as the width.

**Comment by pschlatt1** [comment]
I worked a bit on the intiial condition, and I issued a PR, #1352. You can have a look there and try it out. It should get turbulent for different sizes. You just need to update the domain width in the f90 file, and then also adapt the wavenumbers (probably not even needed).

**Reply by Kamyarmvn** [reply]
Thank you. I will try it out today and report back the results.

**Comment by Kamyarmvn** [comment]
I performed the simulation with the new initial condition and could observe the transition to the turbulent regime. Since I used a finer mesh with a larger time step, I had to reduce the values of eps to avoid divergence. 
Thanks.

**Reply by pschlatt1** [reply]
yes, the amplitudes are quite "aggressvie", so one can certainly reduce them. Here I am not sure whether we even need to random noise, but I prefer to have some level of random noise to break symmetries easier, be it only top-down symmetry.

**Reply by Kamyarmvn** [reply]
Thanks for the help.  I will test the case with the standard EXT case too and report back.

**Comment by Bkv-1915** [comment]
Hi. I am working on turbulent channel flow simulations. I see that the default mesh used in the simulation is non-uniform along x,y and z. I want to know if I can have uniform mesh along x and z direction with non-uniformity along only y direction. Does spectral method allows this, for a polynomial order, say 6 or 8?

**Comment by pschlatt1** [comment]
elements can be spaced uniformely, but the grid points (so-called GLL points) inside the elements are always non-equidistant.

**Reply by Bkv-1915** [reply]
Is there an interpolation tool in nek5000 or neko that can be used for this purpose of transforming the results to uniform grid?

**Comment by pschlatt1** [comment]
there are three ways:
1) use PySEMTools 
2) use internal element-wise interpolation (see the example with TS waves)
3) use global interpolation in Neko

**Reply by Shiyu-Sandy-Du** [reply]
Yes, and for the one using PySEMTools, I have some code for the spectra of a channel: #https://github.com/Shiyu-Sandy-Du/neko_postprocessing/tree/0505c5c020a962c0ef7648f333068866a9b78436/turb_channel_LES. And one needs to change the old keyword "pynektools" into "pysemtools" when importing python modules.
